contracts/starknet/src/tests/test_escrow_claim.cairo

```diff
@@ -0,0 +1,246 @@
+        l1_handler.payload = payload_buffer.span();
+
+        l1_handler.execute().expect('Failed to execute l1_handler');
+
+        // check Order
+        assert(!escrow.get_order_pending(order_id), 'wrong order used');
+        // check balance
+        assert(eth_token.balanceOf(escrow.contract_address) == 0, 'withdraw: wrong balance');
+        assert(eth_token.balanceOf(MM_STARKNET()) == 500, 'withdraw: wrong balance');
+    }
+
+    #[test]
+    fn test_claim_batch_happy() {
+        let (escrow, eth_token) = setup();
+
+        // check balance
+        assert(eth_token.balanceOf(escrow.contract_address) == 0, 'init: wrong balance');
+        assert(eth_token.balanceOf(MM_STARKNET()) == 0, 'init: wrong balance');
+
+        let recipient_addresses = array![12345.try_into().unwrap(), 12346.try_into().unwrap(), 12347.try_into().unwrap()];
+        let amounts = array![500, 501, 502];
+        let fees = array![3, 2, 1];
+
+        let recipient_adresses_clone = recipient_addresses.clone();
+        let amounts_clone = amounts.clone();
+
+        let mut orders = array![];
+
+        start_prank(CheatTarget::One(escrow.contract_address), USER());
+
+        let mut idx = 0;
+        loop {
+            if idx >= 3 {
+                break;
+            }
+
+            let recipient_address = recipient_addresses.at(idx).clone();
+            let amount = amounts.at(idx).clone();
+            let fee = fees.at(idx).clone();
+
+            let order_id = _create_order(recipient_address, amount, fee, escrow);
+
+            orders.append((order_id, recipient_address, amount));
+            
+            idx += 1;
+        };
+
+        // check balance
+        assert(eth_token.balanceOf(escrow.contract_address) == 1509, 'set_order: wrong balance ');
+        assert(eth_token.balanceOf(MM_STARKNET()) == 0, 'set_order: wrong balance');
+
+        // Call withdraw_batch l1_handler
+        let mut l1_handler = L1HandlerTrait::new(
+            contract_address: escrow.contract_address,
+            function_name: 'claim_payment_batch'
+        );
+
+        let mut payload_buffer: Array<felt252> = ArrayTrait::new();
+        Serde::serialize(@orders, ref payload_buffer);
+
+        l1_handler.from_address = ETH_TRANSFER_CONTRACT().into();
+        l1_handler.payload = payload_buffer.span();
+
+        l1_handler.execute().expect('Failed to execute l1_handler');     
+
+        assert(eth_token.balanceOf(escrow.contract_address) == 0, 'withdraw: wrong balance');
+        assert(eth_token.balanceOf(MM_STARKNET()) == 1509, 'withdraw: wrong balance');
+
+        // Check order_ids
+        let mut idx = 0;
+        loop {
+            if idx >= 3 {
+                break;
+            }
+
+            let (order_id, _, _) = orders.at(idx).clone();
+            assert(!escrow.get_order_pending(order_id), 'Order not used');
+            idx += 1;
+        };
+    }
+
+
+    #[test]
+    fn test_fail_random_eth_user_calls_l1_handler() {
```

Comment:
A test like this but for the new claim_payment_batch would be great

Reply:
Added on afa63ea46e63cb1ba712a7337bba490331fda65d